tests/unit/llms/providers/test_watsonx.py

```diff
@@ -40,3 +43,76 @@ def test_basic_interface(provider_config):
     llm = watsonx.load()
     assert isinstance(llm, WatsonxLLM)
     assert watsonx.default_params
+
+
+@patch("ols.src.llms.providers.watsonx.WatsonxLLM", new=WatsonxLLM())
+def test_params_handling(provider_config):
+    """Test that not allowed parameters are removed before model init."""
+    config.init_empty_config()  # needed for checking the config.dev_config.llm_params
+
+    # first two parameters should be removed before model init
+    # rest need to stay
+    params = {
+        "unknown_parameter": "foo",
+        "verbose": True,
+        "min_new_tokens": 1,
+        "max_new_tokens": 10,
+        "temperature": 0.3,
+    }
+
+    watsonx = WatsonX(
+        model="uber-model", params=params, provider_config=provider_config
+    )
+    llm = watsonx.load()
+    assert isinstance(llm, WatsonxLLM)
+    assert watsonx.default_params
+    assert watsonx.params
+
+    # known parameters should be there
+    assert GenParams.DECODING_METHOD in watsonx.params
+    assert watsonx.params[GenParams.DECODING_METHOD] == "sample"
+
+    assert GenParams.MAX_NEW_TOKENS in watsonx.params
+    assert watsonx.params[GenParams.MAX_NEW_TOKENS] == 10
+
+    # unknown parameters should be filtered out
+    assert "unknown_parameter" not in watsonx.params
+    assert "verbose" not in watsonx.params
+
+
+@patch("ols.src.llms.providers.watsonx.WatsonxLLM", new=WatsonxLLM())
+def test_params_handling_none_values(provider_config):
+    """Test that not allowed parameters are removed before model init."""
+    config.init_empty_config()  # needed for checking the config.dev_config.llm_params
+
+    # first three parameters should be removed before model init
+    # rest need to stay
```

Comment:
comment is wrong again?  test seems to assume min_new_tokens/max_new_tokens stays